web/models/assets.ex

```diff
@@ -0,0 +1,154 @@
+defmodule HexWeb.Assets do
+  def push_release(release, body) do
+    opts = [acl: :public_read, cache_control: "public, max-age=604800", meta: [{"surrogate-key", tarball_cdn_key(release)}]]
+    HexWeb.Store.put(nil, :s3_bucket, tarball_store_key(release), body, opts)
+    HexWeb.CDN.purge_key(:fastly_hexrepo, tarball_cdn_key(release))
+  end
+
+  def revert_release(release) do
+    name    = release.package.name
+    version = to_string(release.version)
+
+    # Delete release tarball
+    HexWeb.Store.delete(nil, :s3_bucket, tarball_store_key(release), [])
+
+    # Delete relevant documentation (if it exists)
+    if release.has_docs do
+      HexWeb.Store.delete(nil, :s3_bucket, "docs/#{name}-#{version}.tar.gz", [])
+      paths = HexWeb.Store.list(nil, :docs_bucket, Path.join(name, version))
+      HexWeb.Store.delete_many(nil, :docs_bucket, Enum.to_list(paths), [])
+      HexWeb.Sitemaps.publish_docs_sitemap()
```

Comment:
`HexWeb.Sitemaps.publish_docs_sitemap()` is actually the only call to (under the hood) Repo in this whole module, the rest is just using Store & CDN. Thus, it's a bit different from other "context/crud" modules and so was thinking about putting this as `lib/hex_web/assets.ex`, wdyt? long term everything is gonna be in lib in phoenix 1.3 so perhaps it doesn't matter either way :)